src/DocumentStore.zig

```diff
@@ -156,26 +180,45 @@ fn loadBuildConfiguration(context: LoadBuildConfigContext) !void {
     const global_cache_path = context.global_cache_path;
     const zig_exe_path = context.zig_exe_path;
 
+    var arena = std.heap.ArenaAllocator.init(allocator);
+    defer arena.deinit();
+    const arena_allocator = arena.allocator();
+
     const build_file_path = context.build_file_path orelse try URI.parse(allocator, build_file.uri);
     defer if (context.build_file_path == null) allocator.free(build_file_path);
     const directory_path = build_file_path[0 .. build_file_path.len - "build.zig".len];
 
-    const args: []const []const u8 = &[_][]const u8{
-        zig_exe_path,
-        "run",
-        build_runner_path,
-        "--cache-dir",
-        global_cache_path,
-        "--pkg-begin",
-        "@build@",
-        build_file_path,
-        "--pkg-end",
-        "--",
-        zig_exe_path,
-        directory_path,
-        context.cache_root,
-        context.global_cache_root,
-    };
+    // enough capacity for arguments always passed to the build runner plus a couple build options
+    var arglist = try std.ArrayListUnmanaged([]const u8).initCapacity(arena_allocator, 32);
```

Comment:
You can calculate the exact amount of arguments that are needed and allocate them directly  
```zig
const args = try allocator.alloc([]const u8, 14 + if (build_file.build_options) |opts| opts.len else 0);
defer allocator.free(args);
args[0..14].* = [_][]const u8 {
    // ...
};
if (build_file.build_options) |opts| {
    for (opts) |opt, i| args[14 + i] = opt;
}
```